tools/snapshot_repo_filter.bzl

```diff
@@ -0,0 +1,19 @@
+# Copyright Google Inc. All Rights Reserved.
+#
+# Use of this source code is governed by an MIT-style license that can be
+# found in the LICENSE file at https://angular.io/license
+
+load("//:constants.bzl", "SNAPSHOT_REPOS")
+
+# jq filter that replaces package.json dependencies with snapshot repos
+SNAPSHOT_REPO_JQ_FILTER = _generate_snapshot_repo_filter()
```

Comment:
If preferred, I could just hardcode this into a file. I thought it would be cleaner to generate it from `SNAPSHOT_REPOS`.